src/EnergyPlus/RuntimeLanguageProcessor.cc

```diff
@@ -1272,9 +1288,39 @@ namespace RuntimeLanguageProcessor {
 
 			} else if ( is_any_of( NextChar, "+-*/^=<>@|&" ) ) {
 				// Parse an operator token
-				StringToken = NextChar;
-
-				Token( NumTokens ).Type = TokenOperator;
+				if ( NextChar == '-' ) {
+					StringToken = "-";
+					if ( MultFound ) {
+						ShowSevereError( "EMS Parse Expression, for \"" + ErlStack( StackNum ).Name + "\"." );
+						ShowContinueError( "...Line = \"" + Line + "\"." );
+						ShowContinueError( "...Minus sign used on the right side of multiplication sign." );
+						ShowContinueError( "...Use parenthesis to wrap appropriate variables. For example, X * ( -Y )." );
+						++NumErrors;
+						MultFound = false;
+					} else if ( DivFound ) {
+						ShowSevereError( "EMS Parse Expression, for \"" + ErlStack( StackNum ).Name + "\"." );
+						ShowContinueError( "...Line = \"" + Line + "\"." );
+						ShowContinueError( "...Minus sign used on the right side of division sign." );
+						ShowContinueError( "...Use parenthesis to wrap appropriate variables. For example, X / ( -Y )." );
+						++NumErrors;
+						DivFound = false;
+					} else if ( OperatorProcessing && ( NextChar == '-' ) ) {
+						// if operator was deterined last pass and this character is a -, then insert a 0 before the minus and treat as subtraction
+						// example: change "Var == -1" to "Var == 0-1" 
+						OperatorProcessing = false;
+						String.insert( Pos, "0" );
+						++LastPos;
+						StringToken = "0";
+						MultFound = false;
+						DivFound = false;
```

Comment:
This is the main fix for negatives on right side of IF operator. IF Var == -1, changed to IF Var == 0-1,